x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -150,7 +151,7 @@ export class Session {
   async get(request: KibanaRequest) {
     const sessionCookieValue = await this.options.sessionCookie.get(request);
     if (!sessionCookieValue) {
-      return null;
+      return { error: new SessionMissingError(), value: null };
```

Comment:
Is there a reason you're returning an object containing either an error or the session value here or could we simplify this function to throw the error and return the session value directly?

Reply:
Good question. We've discussed this back and forth several times in the comments above and IRL and I don't think there was _one_ correct answer (previous discussion: https://github.com/elastic/kibana/pull/139811#discussion_r993815965). Throwing the error was one of the options we considered. We also considered using an "Either monad" (even went so far to implement it in https://github.com/watson/kibana/commit/71a13c8854ee9aaeab0d48ad5699a6397d3cf134).

I think we ended up with this implementation because it seemed to be the simplest: A lot of times, the caller doesn't care about the error. They just want to get the session or `null` if it couldn't. Forcing these callers to wrap in a try-catch block seemed like a lot. But I'm happy to change the implementation if you think it's better to throw. The question is then if we should create a `try` wrapper function that returns `null` for those callers who doesn't care about the error, or if all callers should add a try-catch and just ignoring the error.

Reply:
I think returning an object or a tuple is a good approach when you could run into a situation were you need to return both but that doesn't seem to be the case here. 

If the rest of the team prefer this approach happy to keep as is though.

Reply:
I've pushed a commit that throws instead of return a tuple: https://github.com/elastic/kibana/pull/139811/commits/8504abd459f0404e43787f7a8819812d34d13b17

Let me know if you think this is more complicated than the previous implementation. If so, I can just revert it.

Reply:
@thomheymann and I agreed to revert the commit